doc/lfsc1_5.g4

```diff
@@ -89,7 +89,7 @@ type
   | '(' '!' iden ntype type ')' 
 // Extension ///////////////////////////////
   | '(' 'Forall' iden ntype type ')' 
-  | '(' '->' ( ntype | vtype )+ type ')'  
+  | '(' '->' '(' ( ntype | vtype )+ ')' type ')'  
```

Comment:
Surrounding the argument list in parentheses makes it clear when we are done with arguments.
This makes it easier to type-check the return type without constructing an AST for it. Without the parentheses it would be hard to avoid constructing the AST, since we often need an AST for arguments.

Reply:
Also, this makes `->` more like `declare-rule` and `declare-type`.